### ISSUE TITLE
Unable to open code

### ISSUE THREAD

**Issue by lawki** [elementary/code#766]
## Prerequisites
- [✓ ] I have searched open and closed issues for duplicates.

## Describe the bug
Unable to open code. 
When launching from dock, it never opens while the cursor gets replaced by a circular spinner for 10-15 seconds.
When launched from terminal, it gives the following error -
`[GLib-GIO] Settings schema 'io.elementary.code.settings' does not contain a key named 'smart-cut-copy'
[1]    31744 trace trap  io.elementary.code`

This behaviour started about a week ago and meanwhile I've kept the system updated.

#### Method 1
1. Launch Code from dock or application search on top left of desktop

#### Method 2
1. Open terminal,
2. type `io.elementary.code` and press Enter

### Expected behavior
It should launch the text editor

### Screenshots or screen recordings

![The Problem](https://i.imgur.com/siq5ZLt.jpg "The Problem")

### Logs

`#0  0x00007ffff7b0fea1 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#1  0x00007ffff7b111eb in g_logv () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff7b1133f in g_log () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff782d219 in  () at /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
#4  0x00007ffff782d84c in  () at /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
#5  0x00007ffff783136f in g_settings_get_value () at /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
#6  0x00007ffff783246b in g_settings_get_boolean () at /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
#7  0x00007ffff5f5303a in  () at /usr/lib/x86_64-linux-gnu/libgranite.so.5
#8  0x00007ffff5f53cdb in  () at /usr/lib/x86_64-linux-gnu/libgranite.so.5
#9  0x00007ffff74e057e in  () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#10 0x00007ffff74e25c0 in g_object_new_valist () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#11 0x00007ffff74e2939 in g_object_new () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#12 0x00007ffff5f5267f in granite_services_settings_construct () at /usr/lib/x86_64-linux-gnu/libgranite.so.5
#13 0x000055555556ebb8 in scratch_application_construct (object_type=<optimized out>) at ./src/Application.vala:59
#14 0x000055555556ed21 in scratch_application_new () at ./src/Application.vala:46
#15 0x000055555556ed21 in scratch_application_get_instance () at ./src/Application.vala:72
#16 0x000055555556ed80 in scratch_application_main (args=0x7fffffffdd18, args_length1=1) at ./src/Application.vala:257
#17 0x00007ffff4b81b97 in __libc_start_main (main=
    0x55555556dac0 <main>, argc=1, argv=0x7fffffffdd18, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffdd08)
    at ../csu/libc-start.c:310
#18 0x000055555556dafa in _start ()
`

## Platform Information

![System Info](https://i.imgur.com/Apv3SOM.jpg)

- [ ] I'm using the latest version from git that I've manually compiled
- [✓ ] I'm using the latest released stable version

**Comment by danrabbit** [comment]
I can't confirm this issue and as far as I can tell the relevant commit does contain this key: https://github.com/elementary/code/blob/419132ae2d24933456c79cf00251d3f38f0a6f4d/data/io.elementary.code.gschema.xml#L73

**Comment by lawki** [comment]
@danrabbit I've updated my system again today, but same problem. Any logs required from my side that might help diagnose this problem? I'm using Code v3.2.0.

**Comment by danrabbit** [comment]
@lawki can you try reinstalling with `sudo apt install --reinstall io.elementary.code` and see if that resolves the issue for you?

**Comment by lawki** [comment]
@danrabbit it didn't resolve, but it did give a warning on reinstallation
`No such key 'button-layout' in schema 'org.gnome.mutter:Pantheon' as specified in override file '/usr/share/glib-2.0/schemas/20_elementary.pantheon.wm.gschema.override'; ignoring override for this key.
No such key 'enable-animations' in schema 'org.gnome.mutter:Pantheon' as specified in override file '/usr/share/glib-2.0/schemas/20_elementary.pantheon.wm.gschema.override'; ignoring override for this key.
`
PFA the screenshot.
[reinstallation](https://i.imgur.com/DfMNlyg.jpg)

**Comment by FeFiFoFu** [comment]
I have exact same error (e.g., No such key for button-layout and enable-animations), but it's when I use Pantheon Mail.

**Comment by jeremypw** [comment]
@lawki I don't know whether it matters, but I notice you are running code as root.  Have you tried running as a normal user?

**Comment by jeremypw** [comment]
@lawki Do you still experience this after updating to ver 3.3.0 of Code? Make sure there are no remnants of older versions on the system.  Use `io.elementary.code --version` command to check which version is actually going to run.  You can use dconf-editor to look at the settings schemas to check whether the `smart-cut-copy` key was installed.

**Comment by lawki** [comment]
@jeremypw as you can see in first screenshot in the thread, I'm running it via normal user and separately tried opening it from dock. 

As for dconf-editor, I tried resetting values recursively for io.elementary.code but it didn't fix it.
Since then I've moved on to using gedit for now, so I don't know if newer version fixes it.

I'll check on that suggestion and let you know.

**Comment by lawki** [comment]
@jeremypw The newer version doesn't fix it, even though I've installed all available updates. Interestingly on my other system, same version works fine.

I also noticed that `io.elementary.code.settings` doesn't contain `smart-cut-copy` in dconf-editor, even after I reinstalled or uninstalled + purge &rarr; installed. PFA screenshots for reference -

![valid key not found](https://i.imgur.com/OIPrykt.jpg)


![dconf-editor doesn't show the key reported when code fails to open](https://i.imgur.com/qrnNGcZ.jpg)

**Comment by jeremypw** [comment]
@lawki OK, thanks for the extra info. Clearly, the failure of the updated schema to get installed is the cause of the failure to run.  Could you look at the file `/usr/share/glib-2.0/schemas/io.elementary.code.gschema.xml` in a text editor and see whether it contains the missing key?
```
    <key name="smart-cut-copy" type="b">
      <default>true</default>
      <summary>CTRL+C and CTRL+X copy/cut the current line when no selection</summary>
      <description>If there is no selection, will the cut/copy shortcuts automatically act on the current line</description>
    </key>
```
If it does you could try running `sudo glib-compile-schemas /usr/share/glib-2.0/schemas` to see whether any error is incurred or whether it successfully updates the schema database.

**Comment by jeremypw** [comment]
You could also check each folder listed by ` echo $XDG_DATA_DIRS` to see if any contain an old version of the io.elementary.code schema that might be taking precedence.

**Comment by lawki** [comment]
@jeremypw the `schemas` directory does contain that key in exact format as you mentioned, but when I tried compiling the `schemas`, it gave following message -

`No such key 'button-layout' in schema 'org.gnome.mutter:Pantheon' as specified in override file '/usr/share/glib-2.0/schemas/20_elementary.pantheon.wm.gschema.override'; ignoring override for this key.
No such key 'enable-animations' in schema 'org.gnome.mutter:Pantheon' as specified in override file '/usr/share/glib-2.0/schemas/20_elementary.pantheon.wm.gschema.override'; ignoring override for this key.`

I also tried to open code after changing XDG_DATA_DIRS to take /usr/share/pantheon as first in order, but to no avail.

![compiling schemas: failure](https://i.imgur.com/TnCGuvh.jpg)

![XDG_PRECEDENCE](https://i.imgur.com/6ToIkVi.jpg)

**Comment by jeremypw** [comment]
I got the messages about the override files as well - they are not relevant to this issue (I have raised an issue against the relevant project about them).

You probably want `/usr/share` to take precedence (`/usr/share/pantheon` does not exist on my system) but really it would be better to check for old code schema files in the other paths as changing XDG_DATA_DIRS could affect other things. Or else make sure you can revert any changes.

**Comment by lawki** [comment]
@jeremypw the issue is fixed when `/usr/share` takes precedence over other directories. 

As of now that precedence is in my terminal, it's temporary, so  what bad side effects of modifying it for a user's shell be, if I make that precedence order permanent?

**Comment by jeremypw** [comment]
It might affect flatpacks, snap packages or any software that installs stuff below `/usr/loca/` rather than `/usr/`.  Any problem is likely to be reversible by changing $XDG_DATA_DIRS back to the original value. However, you must judge for yourself what risk you are prepared to take depending on what you are using the system for and how much work it would take to reinstall everything. Obviously make sure any valuable data is backed up first.

 You could try changing the precedence in $XDG_DATA_DIRS systematically to identify where the problem lies.  Put `/usr/share/` second and then put first each of the other folders in turn until the problem reoccurs. Then search in and below the problem path for schema related files (`.gschema.xml`, `.gschema.override`, `gschemas.compiled`). Then remove these or move them out of the $XDG_DATA_DIRS paths.

**Comment by amblyline** [comment]
I also have an issue opening Code right now. 
**It just opens up and then closes itself right after.**
I use the most updated version of elementary OS:
![Screenshot from 2020-03-29 17-00-47](https://user-images.githubusercontent.com/36894433/77852422-e234a200-71de-11ea-8ef5-8c99402cb30f.png)

It happened when I tried to open one specific .php file (attached), not earlier.
[buypoints.php.zip](https://github.com/elementary/code/files/4399101/buypoints.php.zip)

**I also tried to open Code app via terminal, feedback was:** 
io.elementary.code

GtkSourceView:ERROR:gtksourcecontextengine.c:5543:update_syntax: assertion failed: (g_slist_length (ce->priv->invalid) <= 1)
Aborted


**Then I tried to reinstall Code, the feedback was:** 
sudo apt install --reinstall io.elementary.code
[sudo] password for zenondanon:         
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libllvm8 libllvm8:i386 linux-headers-5.0.0-37 linux-headers-5.0.0-37-generic
  linux-headers-5.3.0-26 linux-headers-5.3.0-26-generic linux-headers-5.3.0-28
  linux-headers-5.3.0-28-generic linux-image-5.0.0-37-generic
  linux-image-5.3.0-26-generic linux-image-5.3.0-28-generic
  linux-modules-5.0.0-37-generic linux-modules-5.3.0-26-generic
  linux-modules-5.3.0-28-generic linux-modules-extra-5.0.0-37-generic
  linux-modules-extra-5.3.0-26-generic linux-modules-extra-5.3.0-28-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 1 reinstalled, 0 to remove and 0 not upgraded.
Need to get 0 B/426 kB of archives.
After this operation, 0 B of additional disk space will be used.
(Reading database ... 406081 files and directories currently installed.)
Preparing to unpack .../io.elementary.code_3.3.0~r3020+pkg130~ubuntu5.1.2.1_amd64.deb ...
Unpacking io.elementary.code (3.3.0~r3020+pkg130~ubuntu5.1.2.1) over (3.3.0~r3020+pkg130~ubuntu5.1.2.1) ...
Setting up io.elementary.code (3.3.0~r3020+pkg130~ubuntu5.1.2.1) ...
update-alternatives: using /usr/bin/io.elementary.code to provide /usr/bin/gnome-text-editor (gnome-text-editor) in auto mode
Processing triggers for libglib2.0-0:amd64 (2.56.4-0ubuntu0.18.04.6) ...
No such key 'button-layout' in schema 'org.gnome.mutter:Pantheon' as specified in override file '/usr/share/glib-2.0/schemas/20_elementary.pantheon.wm.gschema.override'; ignoring override for this key.
No such key 'enable-animations' in schema 'org.gnome.mutter:Pantheon' as specified in override file '/usr/share/glib-2.0/schemas/20_elementary.pantheon.wm.gschema.override'; ignoring override for this key.
Processing triggers for bamfdaemon (0.5.3+18.04.20180207.2-0ubuntu1) ...
Rebuilding /usr/share/applications/bamf-2.index...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
Processing triggers for gnome-menus (3.13.3-11ubuntu1.1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for mime-support (3.60ubuntu1) ...
Processing triggers for desktop-file-utils (0.23-1ubuntu3.18.04.2+elementary2~ubuntu5.0.1) ...
Processing triggers for libglib2.0-0:i386 (2.56.4-0ubuntu0.18.04.6) ...
No such key 'button-layout' in schema 'org.gnome.mutter:Pantheon' as specified in override file '/usr/share/glib-2.0/schemas/20_elementary.pantheon.wm.gschema.override'; ignoring override for this key.
No such key 'enable-animations' in schema 'org.gnome.mutter:Pantheon' as specified in override file '/usr/share/glib-2.0/schemas/20_elementary.pantheon.wm.gschema.override'; ignoring override for this key.

How do I solve this issue?

**Comment by amblyline** [comment]
**Problem identified:** 
the .php file I attached above (buypoints.php.zip)

I tried to reinstal, purge, update etc the Code app, but nothing worked.
Problem persisted since Code app launched again and again with this .php (recent file) opened, it was causing automatic shutdown of Code app.

I overcame it by just changing this .php file name, so it would not start anymore with Code app.

**To reproduce the problem**, you just need to open this .php file in Code app:
[buypoints.php.zip](https://github.com/elementary/code/files/4399419/buypoints.php.zip)

Any explanation/solution?
